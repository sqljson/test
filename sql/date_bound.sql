SELECT jsonknife_date_bound('1980', 'min');
SELECT jsonknife_date_bound('1980-02', 'min');
SELECT jsonknife_date_bound('1980-02-05','min');
SELECT jsonknife_date_bound('1980-02-05T08', 'min');
SELECT jsonknife_date_bound('1980-02-05T08:30', 'min');

SELECT jsonknife_date_bound(_at, 'min');
SELECT jsonknife_date_bound(_at, 'max');



SELECT jsonknife_date_bound('1980', 'max');
SELECT jsonknife_date_bound('1980-02', 'max');
SELECT jsonknife_date_bound('1980-02-05','max');
SELECT jsonknife_date_bound('1980-02-05T08', 'max');
SELECT jsonknife_date_bound('1980-02-05T08:30', 'max');


select jsonknife_date_bound('2000-01-01', 'min');
select jsonknife_date_bound('2000-01-01', 'max');
