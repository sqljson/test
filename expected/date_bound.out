SELECT jsonknife_date_bound('1980', 'min'); 
     jsonknife_date_bound      
------------------------------
 Tue Jan 01 00:00:00 1980 PST
(1 row)

SELECT jsonknife_date_bound('1980-02', 'min'); 
     jsonknife_date_bound      
------------------------------
 Fri Feb 01 00:00:00 1980 PST
(1 row)

SELECT jsonknife_date_bound('1980-02-05','min'); 
     jsonknife_date_bound      
------------------------------
 Tue Feb 05 00:00:00 1980 PST
(1 row)

SELECT jsonknife_date_bound('1980-02-05T08', 'min'); 
     jsonknife_date_bound      
------------------------------
 Tue Feb 05 08:00:00 1980 PST
(1 row)

SELECT jsonknife_date_bound('1980-02-05T08:30', 'min'); 
     jsonknife_date_bound      
------------------------------
 Tue Feb 05 08:30:00 1980 PST
(1 row)

SELECT jsonknife_date_bound('1980', 'max'); 
         jsonknife_date_bound         
-------------------------------------
 Wed Dec 31 23:59:59.999999 1980 PST
(1 row)

SELECT jsonknife_date_bound('1980-02', 'max'); 
         jsonknife_date_bound         
-------------------------------------
 Fri Feb 29 23:59:59.999999 1980 PST
(1 row)

SELECT jsonknife_date_bound('1980-02-05','max'); 
         jsonknife_date_bound         
-------------------------------------
 Tue Feb 05 23:59:59.999999 1980 PST
(1 row)

SELECT jsonknife_date_bound('1980-02-05T08', 'max'); 
         jsonknife_date_bound         
-------------------------------------
 Tue Feb 05 08:59:59.999999 1980 PST
(1 row)

SELECT jsonknife_date_bound('1980-02-05T08:30', 'max'); 
         jsonknife_date_bound         
-------------------------------------
 Tue Feb 05 08:30:59.999999 1980 PST
(1 row)

select jsonknife_date_bound('2000-01-01', 'min');
     jsonknife_date_bound      
------------------------------
 Sat Jan 01 00:00:00 2000 PST
(1 row)

select jsonknife_date_bound('2000-01-01', 'max');
         jsonknife_date_bound         
-------------------------------------
 Sat Jan 01 23:59:59.999999 2000 PST
(1 row)

